### PR TITLE
Update README with examples of variations on alist structure

### PR DESCRIPTION
**Pull request by Xeijin**
Examples including:

- nested JSON using alist structure
- how to represent a boolean value within the alist structure
- how to selectively evaluate lisp within the alist structure

I struggled to figure these out from the provided documentation, so suggest it's worth including them.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -123,6 +123,23 @@ PUT JSON data::
              (lambda (&key data &allow-other-keys)
                (message "I sent: %S" (assoc-default 'json data)))))
 
+Another PUT JSON example (nested JSON using alist structure, how to represent a boolean & how to selectively evaluate lisp)::
+
+  (request
+   "http://httpbin.org/put"
+   :type "PUT"
+    :data (json-encode `(("jsonArray" . (        ; Prepend alist structure with a backtick (`) rather than single quote (') to allow elisp evaluation of selected elements prefixed with a comma (,)
```
Comment (syohex) [inline README.rst]: Indentation is wrong. And is this comment necessary ? Emacs Lisp programmers understand this thing so I think this comment is redundant.

File: README.rst [hunk 2]
```diff
@@ -123,6 +123,23 @@ PUT JSON data::
...
+                       ("item1" . "value 1")
+                       ("item2" . t) ; This value is expected as a boolean so use the nil / t elisp alist denotation
+                       ("item3" . ,(your-custom-elisp-function)) ; The function will be evaluated as it has been prefixed with a comma (,)
```
Comment (syohex) [inline README.rst]: Ditto.

**Comment by Xeijin** [comment]
I thought it was surprisingly difficult to find out how to do something as simple as POST some nested, pretty-printed JSON that I had, but do take your point about this not really being request.el specific. I find the emacs documentation in general confusing to interpret a lot of the time so maybe it's just me!

**Comment by syohex** [comment]
I have cherry-picked your commit. Thanks

**Pull request closed without merging**